libs/core/algorithms/include/hpx/parallel/algorithms/generate.hpp

```diff
@@ -56,19 +56,50 @@ namespace hpx {
     /// fashion in unspecified threads, and indeterminately sequenced
     /// within each thread.
     ///
-    /// \returns  The \a replace_if algorithm returns a \a hpx::future<FwdIter>
+    /// \returns  The \a generate algorithm returns a \a hpx::future<FwdIter>
     ///           if the execution policy is of type
     ///           \a sequenced_task_policy or
     ///           \a parallel_task_policy
     ///           and returns \a FwdIter otherwise.
-    ///           It returns \a last.
     ///
     template <typename ExPolicy, typename FwdIter, typename F>
     typename util::detail::algorithm_result<ExPolicy, FwdIter>::type
```

Comment:
this should be `typename hpx::parallel::util::detail::algorithm_result<ExPolicy, FwdIter>::type`

same applies to other places as well